### PR TITLE
Add ProgressDeadlineSeconds

### PR DESCRIPTION
**Pull request by dthomson25**
Additionally, I simplified the testing framework to allow developers to test specific fields within the objects sent to the Kubernetes API instead of the entire object.

### CODE REVIEW

File: controller/service.go [hunk 1]
```diff
@@ -114,12 +115,15 @@ func (c *Controller) getPreviewAndActiveServices(r *v1alpha1.Rollout) (*corev1.S
...
 	if r.Spec.Strategy.BlueGreenStrategy.PreviewService != "" {
 		previewSvc, err = c.kubeclientset.CoreV1().Services(r.Namespace).Get(r.Spec.Strategy.BlueGreenStrategy.PreviewService, metav1.GetOptions{})
 		if err != nil {
 			if errors.IsNotFound(err) {
-				logCtx.Warnf("Service %v does not exist", r.Spec.Strategy.BlueGreenStrategy.PreviewService)
+				msg := fmt.Sprintf(conditions.ServiceNotFoundMessage, "preview", r.Spec.Strategy.BlueGreenStrategy.PreviewService)
+				c.recorder.Eventf(r, corev1.EventTypeWarning, conditions.ServiceNotFoundReason, msg)
+				cond := conditions.NewRolloutCondition(v1alpha1.RolloutProgressing, corev1.ConditionFalse, conditions.ServiceNotFoundReason, msg)
+				conditions.SetRolloutCondition(&r.Status, *cond)
+				c.rolloutsclientset.ArgoprojV1alpha1().Rollouts(r.Namespace).Update(r)
```
Comment (alexmt) [inline controller/service.go]: I think we should use Patch to set conditions.

Comment (alexmt) [inline controller/service.go]: Return Update/Patch error

File: controller/service.go [hunk 2]
```diff
@@ -130,7 +134,11 @@ func (c *Controller) getPreviewAndActiveServices(r *v1alpha1.Rollout) (*corev1.S
 	activeSvc, err = c.kubeclientset.CoreV1().Services(r.Namespace).Get(r.Spec.Strategy.BlueGreenStrategy.ActiveService, metav1.GetOptions{})
 	if err != nil {
 		if errors.IsNotFound(err) {
-			logCtx.Warnf("Service %v does not exist", r.Spec.Strategy.BlueGreenStrategy.PreviewService)
+			msg := fmt.Sprintf(conditions.ServiceNotFoundMessage, "active", r.Spec.Strategy.BlueGreenStrategy.ActiveService)
```
Comment (alexmt) [inline controller/service.go]: Same as for line 126

File: utils/conditions/conditions.go [hunk 3]
```diff
@@ -40,6 +42,70 @@ const (
 	DuplicatedServicesMessage = "This rollout uses the same service for the active and preview services, but two different services are required."
 	// Available the reason to indicate that the rollout is serving traffic from the active service
 	Available = "Available"
+
+	// Reasons and Messages for rollout conditions
+
+	// Progressing:
```
Comment (alexmt) [inline utils/conditions/conditions.go]: Delete this line?

File: controller/service.go (hunk 1 above)
Comment (dthomson25) [reply controller/service.go]: 👍

Comment (dthomson25) [reply controller/service.go]: 👍

File: controller/service.go (hunk 2 above)
Comment (dthomson25) [reply controller/service.go]: 👍

File: utils/conditions/conditions.go (hunk 3 above)
Comment (dthomson25) [reply utils/conditions/conditions.go]: 👍

**Pull request merged**